### PR TITLE
Update variation synonyms script

### PR DESCRIPTION
**Pull request by dglemos**
Add UniProt synonyms to variation_set_variation: [ENSVAR-3720](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-3720)

Support EVA synonyms: [ENSVAR-4968](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-4968)
Support Pig chip Consortium: [ENSVAR-4969](https://www.ebi.ac.uk/panda/jira/browse/ENSVAR-4969)

### CODE REVIEW

File: scripts/import/import_variant_synonyms [hunk 1]
```diff
@@ -165,6 +167,45 @@ elsif($source_name eq 'UniProt'){
...
+    status      => "germline",
+  }, 
+  );
+
+  # get source info 
+  $source_description ||= $source_eva{$source_name}->{description};
+  $source_url ||= $source_eva{$source_name}->{url};
+  my $source_status = $source_eva{$source_name}->{status};
+
+  my $source = get_source($species, $dba, $source_name, $source_version, $source_url, $source_description, $source_status);
```
Comment (nuno-agostinho) [inline scripts/import/import_variant_synonyms:187]: `get_source()` is hardcoded for human. Maybe we should replace [line 635](https://github.com/Ensembl/ensembl-variation/blob/b5c2fcc0ee634f2cb1996b1b25b0ff262bb12a02/scripts/import/import_variant_synonyms#L635) with:
```perl
my $source_adaptor = $dba->get_SourceAdaptor($species, 'variation', );
```

There are other hardcoded mentions to human in this script, maybe we should use `$species` where possible instead?

File: scripts/import/import_variant_synonyms [hunk 2]
```diff
@@ -165,6 +167,45 @@ elsif($source_name eq 'UniProt'){
...
+  # Source Pig SNP Consortium
+  my %source_data = (
+    "name"           => "Pig SNP Consortium",
+    "version"        => undef,
+    "description"    => "PorcineSNP60 BeadChip",
+    "url"            => "https://www.animalgenome.org/repository/pig/",
+    "somatic_status" => "germline"
+  );
+
+  my $source = get_source($species, $dba, $source_data{name}, $source_data{version}, $source_data{url}, $source_data{description}, $source_data{somatic_status});
```
Comment (nuno-agostinho) [inline scripts/import/import_variant_synonyms:204]: Why not allow to override parameters here as well (e.g. description, url, status), like in the case of EVA?

File: scripts/import/import_variant_synonyms [hunk 3]
```diff
@@ -165,6 +167,45 @@ elsif($source_name eq 'UniProt'){
...
+  $source_description ||= $source_eva{$source_name}->{description};
+  $source_url ||= $source_eva{$source_name}->{url};
+  my $source_status = $source_eva{$source_name}->{status};
+
+  my $source = get_source($species, $dba, $source_name, $source_version, $source_url, $source_description, $source_status);
+
+  import_synonyms($result_synonyms, $source, $dba, $species);
+}
+
+elsif($source_name eq 'pig_chip') {
```
Comment (nuno-agostinho) [inline scripts/import/import_variant_synonyms:192]: Why having a specific parameter for pig? Can't this be generalised for any species — if needed for the future?

File: scripts/import/import_variant_synonyms [hunk 4]
```diff
@@ -433,6 +485,44 @@ sub extract_PharmGKB {
...
+  }
+
+  # Read through the file and parse out the desired fields: 
+  # current rs id and synonym
+  while (my $row = <$fh>) {
+    chomp $row;
+    next if($row =~ /^#/);
+    
+    my @row_data = split /\t/, $row;
+    my $synonym = $row_data[2];
```
Comment (nuno-agostinho) [inline scripts/import/import_variant_synonyms:515]: Should we use the same variable name for different variable types (`$synonym`, `%synonym`)?

File: scripts/import/import_variant_synonyms (hunk 3 above)
Comment (dglemos) [reply scripts/import/import_variant_synonyms:192]: This method is getting data only from pig_synonym, it's specific to pig only.

File: scripts/import/import_variant_synonyms (hunk 1 above)
Comment (dglemos) [reply scripts/import/import_variant_synonyms:187]: I updated line 635.
The other mention is in the dbSNP source which is specific to human.

File: scripts/import/import_variant_synonyms (hunk 4 above)
Comment (dglemos) [reply scripts/import/import_variant_synonyms:515]: Shouldn't use the same, it's updated now.

File: scripts/import/import_variant_synonyms [hunk 5]
```diff
@@ -615,7 +737,7 @@ sub usage{
 
   die "\nUsage : import_variant_synonyms -registry [registry file] -source_name [name]
 \n\tOptions:
-\t         -data_file          [name of file to load]
+\t         -data_file          [name of file to load]  - mandatory for EVA
 \t         -source_version     [version]
```
Comment (nuno-agostinho) [inline scripts/import/import_variant_synonyms:741]: Add `source_name` and list of possible sources to usage options.

```suggestion
\t         -source_name        [source] - PharmGKB , dbSNP, UniProt, EVA or pig_chip
\t         -source_version     [version]
```

**Review by nuno-agostinho — COMMENTED** [review]
Also tested importing from UniProt: the `source`, `variation_synonym`, `variation_set` and `variation_set_variation` tables seem fine.

**Review by nuno-agostinho — APPROVED** [review]

**Comment by nuno-agostinho** [comment]
Merged to release/108 and main.

**Pull request merged**